W-14795609 Connector ACB Configuration (#2478)
mulesoft/docs-connectors: connectors-home/modules/ROOT/partials/configure-acb-connectors-partial.adoc

:page-deployment-options: cloud-ide, desktop-ide
:page-aliases: int-create-basic-integrations.adoc

include::reuse::partial$beta-banner.adoc[tag="anypoint-code-builder"]

//partial used in ACB connector configuration 
//LOGO (web, desktop, or both)
// include::partial$acb-ide-logos.adoc[tags="both-ides"]

// tag::acb-connector-config-intro[]
include::reuse::partial$beta-banner.adoc[tag="anypoint-code-builder"]

Create a basic integration project in Anypoint Code Builder, add connectors and other components to your Mule application to process your business logic, and configure attributes. 

== Before You Begin

Before creating an integration project, you must:

> * xref:anypoint-code-builder::start-acb.adoc[Set up and access the Anypoint Code Builder web or desktop IDE].

Reviewer: ```suggestion
* xref:anypoint-code-builder::start-acb.adoc[Set up and access the Anypoint Code Builder web or desktop IDE]
```

+
If the connector is not available locally, click the image:anypoint-code-builder::icon-search-exchange.png["",15,15] (*Search in Exchange*) toggle:
+
image:anypoint-code-builder::int-connector-search-exchange.png["Search in Exchange toggle"]
+
--
[calloutlist]
. Search locally
. Search in Exchange
--
.. Select the connector to add to your project.
.. Select the operation from the *Add Component* panel.

Adding a connector to an integration project does not make that connector available to other projects in your Anypoint Code Builder workspace.
// end::acb-add-connector-to-project[]

[[configure-source]]
== Configure a Source for the Flow
// tag::acb-configure-source[]
A source (trigger) starts the flow when specified conditions are met. If a connector has its own associated sources, you can use a connector-specific source to initiate the flow. Each connector source has a unique configuration.

You can configure the following sources for this connector:

include::connectors::partial$t-studio-partial.adoc[tags=listener-scheduler-descriptions]

For example, to configure an HTTP Listener source, follow these steps:

. Click the *Listener* component on the canvas.

> . (Optional) Click the edit icon to change the name of the Listener component. 

Reviewer: ```suggestion
. (Optional) Click the edit icon to change the name of the *Listener* component. 
```

. (Optional) If you created a configuration file for the connector, select it. 
. In the *General* tab, in the *Path* field, enter the relative path for the path set in the HTTP configuration element. See xref:http-connector::http-listener-ref.adoc#configure-paths[Configure Paths]. 
. Select the *Advanced* tab to configure the following attributes:
+
[%header%autowidth.spread]
|===
| UI Field | Description 
|Allowed methods |To allow all HTTP methods (default), do not define this attribute. To restrict certain methods, specify all allowed HTTP methods, for example, `GET, POST`, as a comma-separated list.  
| Response streaming mode | Defines whether or not to use streaming when the response is sent.
| Output MIME type | MIME type of the payload that this component outputs. 
|Parameters | Appends a key-value pair for a reader property to the value of the Output MIME attribute. Multiple key-value pairs are allowed.  
| Output encoding | Encoding of the payload that this component outputs.

> |Streaming Strategy | This attribute is not currently enabled in the configuration UI. Configure it in the XML editor. See xref:connectors::partial$streaming-strategies.adoc[].

Reviewer: ```suggestion
|Streaming Strategy | This attribute is not currently enabled in the configuration UI. Configure it in the XML editor. See include::connectors::partial$streaming-strategies.adoc[].
```

> |Primary node only |Select this option to execute the *Listener* trigger on the primary node only when running in a cluster.

Reviewer: ```suggestion
|Primary node only |Select this option to execute the *Listener* source on the primary node only when running in a cluster.
```

Author: ACB uses trigger, so I'm aligning with ACB terminology.

|Redelivery policy |This attribute is not currently enabled in the configuration UI. Configure it in the XML editor. See xref:http-documentation.adoc#RedeliveryPolicy[Redelivery Policy].

> |Reconnection strategy |This attribute is not currently enabled in the configuration UI. Configure it in the XML editor. See xref:http-documentation.adoc#reconnect[Reconnect] and xref:http-documentation.adoc#reconnect[Reconnect Forever].

Reviewer: ```suggestion
|Reconnection Strategy |This attribute is not currently enabled in the configuration UI. Configure it in the XML editor. See xref:database-documentation.adoc#reconnect[Reconnect], xref:database-documentation.adoc#reconnect[Reconnect Forever], and <<configuring-reconnection-strategy>>.
```

> | Response |See xref:http-documentation.adoc#HttpResponseAttributes[HTTP Response Attributes]

Reviewer: ```suggestion
| Response |See xref:http-documentation.adoc#HttpResponseAttributes[HTTP Response Attributes].
```

| Error Response | HTTP error response returns a status code for the error. 
|Body | Body of the error message.
|Headers |Headers to include in the error message.

> |Status code | Status code of the error response, for example `500`.

Reviewer: ```suggestion
|Status code | Status code of the error response, for example, `500`.
```

+
.. Type `http`, and select the `http:listener-config` snippet:
+
image:anypoint-code-builder::add-http-config-snippet.png["http:listener-config snippet highlighted in the configuration XML menu"]
+
The snippet adds the following code:
+
[source,xml]
----
<http:listener-config name="HTTP_Listener_config" >
  <http:listener-connection host="0.0.0.0" port="8081" />
</http:listener-config>
----
// end::acb-add-global-config[]

== Configure the Component Attributes
// tag::acb-configure-attributes[]

After you add components to your integration project, configure the attributes in the component UI, or configure the attributes directly in the XML code editor. 

Connector XML tags start with a namespace followed by an operation, following the syntax `namespace:operation`, such as `<db:select/>` for a *Select* operation in Anypoint Connector for Database (Database Connector).
// end::acb-configure-attributes[]

[[create-new-config]]
== Create a New Configuration XML File
// tag::acb-create-config-file-acb[]

Anypoint Code Builder creates the initial Mule configuration XML file in `src/main/mule`
within the Mule project.
This file contains global configurations and flow structures.

> You can create additional configuration files, for example to store global configurations.

Reviewer: ```suggestion
You can create additional configuration files, for example, to store global configurations.
```

:page-deployment-options: cloud-ide, desktop-ide
:page-aliases: int-create-basic-integrations.adoc

include::reuse::partial$beta-banner.adoc[tag="anypoint-code-builder"]

//partial used in ACB connector configuration 
//LOGO (web, desktop, or both)
// include::partial$acb-ide-logos.adoc[tags="both-ides"]

// tag::acb-connector-config-intro[]
include::reuse::partial$beta-banner.adoc[tag="anypoint-code-builder"]

Create a basic integration project in Anypoint Code Builder, add connectors and other components to your Mule application to process your business logic, and configure attributes. 

== Before You Begin

Before creating an integration project, you must:

* xref:anypoint-code-builder::start-acb.adoc[Set up and access the Anypoint Code Builder web or desktop IDE].

> * Have credentials to access the Connector's API

Reviewer: ```suggestion
* Have credentials to access the connector's API
```

To use this connector, you must be familiar with:

> * The Connector’s API

Reviewer: ```suggestion
* The connector’s API
```

+
image:anypoint-code-builder::int-connector-search-exchange.png["Search in Exchange toggle"]
+
--
[calloutlist]
. Search locally
. Search in Exchange
--
.. Select the connector to add to your project.
.. Select the operation from the *Add Component* panel.

Adding a connector to an integration project does not make that connector available to other projects in your Anypoint Code Builder workspace.
// end::acb-add-connector-to-project[]

[[configure-source]]
== Configure a Source for the Flow
// tag::acb-configure-source[]
A source (trigger) starts the flow when specified conditions are met. If a connector has its own associated sources, you can use a connector-specific source to initiate the flow. Each connector source has a unique configuration.

You can configure the following sources for this connector:

include::connectors::partial$t-studio-partial.adoc[tags=listener-scheduler-descriptions]

For example, to configure an HTTP Listener source, follow these steps:

. Click the *Listener* component on the canvas.
. (Optional) Click the edit icon to change the name of the Listener component. 
. (Optional) If you created a configuration file for the connector, select it. 

> . In the *General* tab, in the *Path* field, enter the relative path for the path set in the HTTP configuration element. See xref:connectors::http-listener-ref.adoc#configure-paths[Configure Paths]. 

Reviewer: this xref doesn't work

| Response streaming mode | Defines whether or not to use streaming when the response is sent.
| Output MIME type | MIME type of the payload that this component outputs. 
|Parameters | Appends a key-value pair for a reader property to the value of the Output MIME attribute. Multiple key-value pairs are allowed.  
| Output encoding | Encoding of the payload that this component outputs.
|Streaming Strategy | This attribute is not currently enabled in the configuration UI. Configure it in the XML editor. See xref:connectors::streaming-strategies.adoc[].
|Primary node only |Select this option to execute the *Listener* trigger on the primary node only when running in a cluster.
|Redelivery policy |This attribute is not currently enabled in the configuration UI. Configure it in the XML editor. See xref:http-documentation.adoc#RedeliveryPolicy[Redelivery Policy].
|Reconnection strategy |This attribute is not currently enabled in the configuration UI. Configure it in the XML editor. See xref:http-documentation.adoc#reconnect[Reconnect] and xref:http-documentation.adoc#reconnect[Reconnect Forever].

> | Response |See xref:http-documentation.adoc#HttpResponseAttributes[HTTP Respoonse Attributes]

Reviewer: ```suggestion
| Response |See xref:http-documentation.adoc#HttpResponseAttributes[HTTP Response Attributes].
```

| Error Response | HTTP error response returns a status code for the error. 
|Body | Body of the error message.
|Headers |Headers to include in the error message.

> |Status code | Status code of the error response, for example `500`.

Reviewer: ```suggestion
|Status code | Status code of the error response, for example, `500`.
```

> |Reason phrase | Reason for the error, for example `Gateway Timeout` or `Internal Server Error`.

Reviewer: ```suggestion
|Reason phrase | Reason for the error, for example, `Gateway Timeout` or `Internal Server Error`.
```
